File: BinomialCoefficient/C#/BinomialCoefficient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Numerics;

namespace BinomialCoefficient
{
	class Program
	{
		public static BigInteger factorial(BigInteger N)
		{
			
			if(N == 0 || N == 1)
				return 1;

			BigInteger result = 1;
			

			for(BigInteger i = 2; i < N + 1; i++)
			{	
				result *= i;
			}

			return result;
		}

		public static BigInteger binomialCoefficient(BigInteger n, BigInteger k)
		{
			BigInteger result = 1;
			
			result = factorial(n) / (factorial(n - k) * factorial(k));

			return result;

		}
		
		//https://stackoverflow.com/questions/9619743/how-to-calculate-binomial-coefficents-for-large-numbers
		public static BigInteger binomialCoefficientStack(BigInteger n, BigInteger k)
		{
			BigInteger result = 1;
			if(k > n - k)
				k = n - k;

			for(BigInteger i = 1; i < k + 1; i++)
			{
				result *= n - k + i;
				result /= i;
			}

			return result;
		}

		public static void Main(string[] args)
		{
			Console.WriteLine(binomialCoefficient(40, 20));
		 	Console.WriteLine(binomialCoefficientStack(40, 20));	
		}
	}
}

File: BinomialCoefficient/Python/BinomialCoefficient.py
def factorial(N):

    if (N == 1 or N == 0):
        
        return 1
    
    elif (N > 1):
    
        result = 1

        for i in range(1, N + 1):
            result *= i
        
        return result

    elif (N < 0):
        return "Error"



def binomialCoefficient(listOfCoefficients):
    
    n = listOfCoefficients[0] 
    k = listOfCoefficients[1] 

    result = factorial(n) / (factorial(k) * factorial(n - k))

    return result
    

listOne = [7, 2]
listTwo = [40, 20]

print(binomialCoefficient(listOne))
print(binomialCoefficient(listTwo))
